graphql/handler/transport/options.go

```diff
@@ -18,9 +22,16 @@ func (o Options) Supports(r *http.Request) bool {
 func (o Options) Do(w http.ResponseWriter, r *http.Request, exec graphql.GraphExecutor) {
 	switch r.Method {
 	case http.MethodOptions:
-		w.Header().Set("Allow", "OPTIONS, GET, POST")
+		w.Header().Set("Allow", o.allowedMethods())
 		w.WriteHeader(http.StatusOK)
 	case http.MethodHead:
 		w.WriteHeader(http.StatusMethodNotAllowed)
 	}
 }
+
+func (o Options) allowedMethods() string {
+	if o.AllowedMethods == nil {
```

Comment:
According to [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231#section-7.4.1), `Allow` header can be empty, I think it needs to distinguish between `nil` (= default header) and `[]string{}` (= empty) header, although I have no idea when a GraphQL server will return empty `Allow` header.

Reply:
I can't think of how it would be beneficial to any use case to return an empty `Allow` header, and it seems much more likely it could be potentially break existing usage since the zero value of the Options would then return an empty `Allow`.

Reply:
I would like to commit the suggestion by @giautm and then merge this, but I will wait to see if @morikuni can explain how this won't cause issues to existing users as is.

Reply:
I agree that empty `Allow` header may not have any benefit for general usage of GraphQL server, so I don't worry to accept the suggestion by @giautm. It's up to you guys whether `gqlgen` will support RFC correctly or support general usage with preventing the mistake of empty `Allow` header.

From the existing user's point of view, both codes (`== nil` and `len != 0`) behave as the same because no users have  set the newly added `AllowedMethods` field and it will fallback to the current `Allow` header (`"OPTIONS, GET, POST"`).

Reply:
Fair enough. I will merge it as is, but if we get user reports of problems, we can change it then.